### PR TITLE
refac: findValidatedForcedDecision method moved to decision service

### PR DESCRIPTION
**Pull request by mnoman09**
## Summary
-  Moves findValidatedForcedDecision method to decision service as validatedForcedDecision

## Test plan
All unit test and FSC tests should pass.

### CODE REVIEW

File: packages/optimizely-sdk/lib/optimizely_user_context/index.ts [hunk 1]
```diff
@@ -213,91 +213,6 @@ export default class OptimizelyUserContext {
...
-        }
-      }
-    }
-
-    return {
-      result: variation,
-      reasons: decideReasons,
-    }
-  }
-
```
Comment (ozayr-zaviar) [inline packages/optimizely-sdk/lib/optimizely_user_context/index.ts:300]: import logger on line 16 and `get_logger` on line 35 can be removed.

**Review by msohailhussain — COMMENTED** [review]
lgtm

**Review by jaeopt — APPROVED** [review]
LGTM

**Review by jaeopt — APPROVED** [review]
LGTM

File: packages/optimizely-sdk/lib/core/decision_service/index.ts [hunk 2]
```diff
@@ -776,6 +776,98 @@ export class DecisionService {
...
+    flagKey: string,
+    ruleKey?: string
+  ): DecisionResponse<Variation | null> {
+
+    const decideReasons: (string | number)[][] = [];
+    const forcedDecision = user.getForcedDecision({ flagKey, ruleKey });
+    let variation = null;
+    let variationKey;
+    const userId = user.getUserId()
+    if (config && forcedDecision) {
```
Comment (zashraf1985) [inline packages/optimizely-sdk/lib/core/decision_service/index.ts:799]: Suggestion (non binding): If we move the check for `config` at the top, we will save a redundant call to `getForcedDecision` when config is not available.

**Review by zashraf1985 — APPROVED** [review]
LGTM! with a non binding suggestion.

**Pull request merged**